src/client/apps/edit/components/content/sections/news/EditSourceControls.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import { borderedInput } from '@artsy/reaction/dist/Components/Mixins'
+import Button from "@artsy/reaction/dist/Components/Buttons/Default"
+
+interface Props {
+  onApply: (news_source: { title?: string, url?: string } | null) => void
+  source?: {
+    title: string
+    url: string
+  }
+}
+
+interface State {
+  title?: string
+  url?: string
+}
+
+export class EditSourceControls extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    const source = props.source
+
+    this.state = { 
+      title: source ? source.title : "",
+      url: source ? source.url : ""
+    }
+  }
+
+  render() {
+    return (
+      <EditSourceContainer>
+        <InputContainer>
+        <Input
+          name={"title"}
+          value={this.state.title}
+          placeholder={"Enter source name"}
+          onChange={(event) => this.setState({title: event.target.value})}
+        />
+        <ApplyInputContainer>
+          <LinkInput
+            name={"url"}
+            value={this.state.url}
+            placeholder={"Paste or type a link"}
+            onChange={(event) => this.setState({url: event.target.value})}
+          />
```

Comment:
Is it possible to do some kind of validation here to make sure it is a valid url?

Reply:
The use case i'm thinking of is for having/not having http://, utm or something like that -- I know we do a quick check for links being valid in body text here-- https://github.com/artsy/positron/blob/master/src/api/apps/articles/model/save.coffee#L166
maybe something along these lines, or sanitize the link on the save side.